message_ix/tests/test_feature_duration_time.py

```diff
@@ -0,0 +1,288 @@
+        for (h_in, h_act) in time_pairs:
+            # "input"
+            inp = com_dict[tec]["input"]
+            if inp:
+                inp_spec = [yr, yr, "standard", "fairyland", inp, "final", h_act, h_in]
+                scen.add_par("input", ["fairyland", tec] + inp_spec, 1, "-")
+        # "output"
+        for h in times_out:
+            out = com_dict[tec]["output"]
+            out_spec = [yr, yr, "standard", "fairyland", out, "final", h, h]
+            scen.add_par("output", ["fairyland", tec] + out_spec, 1, "-")
+
+    # Committing
+    scen.commit("scenario was set up.")
+
+    # Testing if the model solves in GAMS
+    scen.solve(case=comment)
+
+    # Testing if sum of duration_time is almost 1
+    for tmp_lvl in scen.set("lvl_temporal"):
+        times = scen.set("map_temporal_hierarchy", {"lvl_temporal": tmp_lvl})[
+            "time"
+        ].to_list()
+        print(sum(scen.par("duration_time", {"time": times})["value"]))
+        assert (
+            abs(sum(scen.par("duration_time", {"time": times})["value"]) - 1.0) < 1e-12
+        )
+    return scen
+
+
+# Tests for "duration_time" of different number of time slices, at different
+# temporal levels ("lvl_temporal")
+# In these tests "demand" is defined in different time slices and one power plant
+# to meet demand, which receives fuel from a supply technology.
+
+# 1) Testing one temporal level ("season") and different number of time slices
+def test_season(test_mp, n_time=[4, 12, 50]):
+    comment = "season"
+    com_dict = {"power-plant": {"input": [], "output": "electr"}}
+    # Dictionary of technology and input and output temporal levels
+    tec_time = {"power-plant": ["season", "season"]}
+    # Total demand in a temporal level
+    demand_time = {"season": 100}
+    # List of info for time slices: [temporal level, number, parent level]
+    for t in n_time:
+        time_steps = [["season", t, "year"]]
+        # Check the model solves without error and sum of duration times = 1
+        model_generator(test_mp, comment, tec_time, demand_time, time_steps, com_dict)
+
+
+@pytest.mark.skipif(
+    condition="GITHUB_WORKFLOW" in os.environ,
+    reason="Exceeds maximum size for GAMS without license.",
+)
+# 2) Testing one temporal level ("season") linked to "year" with a technology
+def test_year_season(test_mp, n_time=[50, 365, 2400]):
+    comment = "year_season"
+    com_dict = {
+        "power-plant": {"input": "fuel", "output": "electr"},
+        "fuel-supply": {"input": [], "output": "fuel"},
+    }
+    # Dictionary of technology and temporal level
+    tec_time = {"power-plant": ["year", "season"], "fuel-supply": ["year", "year"]}
+    # Total demand in a temporal level
+    demand_time = {"season": 100}
+    # List of info for time slices: [temporal level, number, parent level]
+    for t in n_time:
+        time_steps = [["season", t, "year"]]
+        # Check the model solves without error and sumof duration times = 1
```

Comment:
```suggestion
    # Check the model solves without error and the sum of duration_time = 1
```
Or
`    # Check the model solves without error and the sum of duration time equals one`